queue/queue.go

```diff
@@ -173,6 +188,41 @@ func (q *queue) Exists() (bool, error) {
 	return exist, nil
 }
 
+// Identify to a shared session.
+// In the queue the session has a unique UUID and many connections may share
+// one logical session. Also, the consumer can reconnect to the existing
+// session during the ttr time.
+// To get the UUID of the current session, call the Queue.Identify(nil).
+func (q *queue) Identify(u *uuid.UUID) (*uuid.UUID, error) {
```

Comment:
Identify returns pointer, why?

Reply:
You are right, it would be better to return a value [in this case](https://pkg.go.dev/github.com/google/uuid#UUID).

We can replace a pointer with a value for the argument as well (and add some function to register the session), but it will not look like the Lua api.

Updated.